gen/zap.go

```diff
@@ -86,6 +86,27 @@ func (z *zapGenerator) zapEncoder(g Generator, spec compile.TypeSpec) string {
 	panic(root)
 }
 
+// zapTypedefHasGeneratedMarshaler defines if a typedef will have generated code
+// zapcore.ObjectMarshaler or zapcore.ArrayMarshaler.
+func (z *zapGenerator) zapTypedefHasGeneratedMarshaler(g Generator, spec compile.TypeSpec) bool {
+	return z.zapEncoder(g, spec) == "Array" || z.zapEncoder(g, spec) == "Object"
+}
+
+// zapTypedefGenerateMarshaler is similar to zapMarshaler but always converts a typedef
+// to its root value. This should be used when generating the code for the Zap
+// marshal implementation of the typedef.
+func (z *zapGenerator) zapTypedefGenerateMarshaler(g Generator, spec compile.TypeSpec, fieldValue string) (string, error) {
+	if _, ok := spec.(*compile.TypedefSpec); !ok {
+		panic(spec)
```

Comment:
Seeing as this panics by default anyway without the `!ok` form and we're not
providing any additional context, we can just let it panic on the cast and get
the coverage back.

```suggestion
	_ = spec.(*compile.TypedefSpec)
```